src/tree.rs

```diff
@@ -0,0 +1,59 @@
+//! The baseline requirements of any UI Tree so Taffy can efficiently calculate the layout
+
+use crate::{
+    layout::{Cache, Layout},
+    prelude::*,
+};
+
+/// Any item that implements the LayoutTree can be layed out using Taffy's algorithms.
+///
+/// Generally, Taffy expects your Node tree to be indexable by stable indicies. A "stable" index means that the Node's ID
+/// remains the same between re-layouts.
+pub trait LayoutTree {
+    /// Get the list of children IDs for the given node
+    fn children(&self, node: Node) -> &[Node];
```

Comment:
Maybe here you restrict too much the type of the underlying `children` set (which now *has* to be a `Vec` or something like that) and it's be better to have something like
```rust
     fn children(&self, node: Node) -> impl Iterator<Item=Node>;
```
(which also has a trivial implementation). Now if the `LayoutTree` implementor uses e.g. a `BTreeMap` as the underlying storage it still can trivially implement `children()`.

Reply:
Wouldn't `IntoIterator` consume the collection (`into_iter()` takes `self` by value) ? I think we just want using `iter()` here.